compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -495,6 +496,29 @@ class Semantic {
           Result(value2, errors)
       }
     }
+
+    def accessLocal(sym: Symbol, default: Result, klass: ClassSymbol, source: Tree): Contextual[Result] =
+      sym.defTree match {
+        case vdef: ValDef =>
+          // resolve this for local variable
+          val enclosingClass = sym.owner.enclosingClass.asClass
+          val thisValue2 = resolveThis(enclosingClass, value, klass, source)
+          thisValue2 match {
+            case Hot => Result(Hot, Errors.empty)
+            
+            case Cold => 
+              val error = AccessCold(sym, source, trace.toVector)
+              Result(Hot, error :: Nil)
+            
+            case addr: Addr => 
+              val res = eval(vdef.rhs, addr, klass)
+              if res.value.promote("Try promote", source).isEmpty then Result(Hot, Errors.empty) else res
+                
+            case _ => ???
```

Comment:
Let's print some debug information here: `report.error(...)`. We can define a helper method:

```scala
def reportImpossibleError(message: String)(using Trace, Context): Unit = ...
```

For end-users, this is better than a compiler crash and useful debug information can be reported.

This can be done in another PR.